src/pe/resource.rs

```diff
@@ -0,0 +1,1922 @@
+use crate::error;
+use alloc::string::{String, ToString};
+use alloc::vec::Vec;
+use core::fmt;
+use core::ops::Not;
+use log::debug;
+use scroll::ctx;
+use scroll::{Pread, Pwrite, SizeWith};
+
+use crate::pe::data_directories;
+use crate::pe::options;
+use crate::pe::section_table;
+use crate::pe::utils;
+
+/// Size of `wchar_t` in C (aka [`u16`] in Rust)
+pub(super) const SIZE_OF_WCHAR: usize = core::mem::size_of::<u16>();
+
+/// Converts [`u8`] slice into a vector of [`u16`] and then utf-16 [`String`].
+pub(super) fn to_utf16_string(bytes: &[u8]) -> Option<String> {
+    if bytes.len() % 2 != 0 {
+        return None;
+    }
+    let u16_chars = bytes
+        .chunks_exact(2)
+        .map(|chunk| u16::from_le_bytes([chunk[0], chunk[1]]))
+        .take_while(|&wchar| wchar != 0)
+        .collect::<Vec<_>>();
+    Some(String::from_utf16_lossy(&u16_chars))
+}
+
+/// Helper for parsing `wchar_t` utf-16 strings in a safe manner.
+#[derive(Debug, Copy, Clone, Default, PartialEq)]
+pub struct Utf16String<'a>(&'a [u8]);
```

Comment:
this is still `pub`; since key is now `pub(crate)` we can do the same and then we can merge. Thank you for removing the unsafe  code!